src/gsym.rs

```diff
@@ -0,0 +1,90 @@
+use std::fs::File;
+use std::io::{Error, Read};
+use std::mem;
+use std::path::PathBuf;
+
+use super::{AddressLineInfo, FindAddrOpts, SymResolver, SymbolInfo};
+
+mod parser;
+mod types;
+
+use parser::{find_address, GsymContext};
+
+/// The symbol resolver for the GSYM format.
+pub struct GsymResolver {
+    file_name: PathBuf,
+    ctx: GsymContext<'static>,
+    #[allow(dead_code)]
+    data: Vec<u8>,
+    loaded_address: u64,
+}
+
+impl GsymResolver {
+    pub fn new(file_name: PathBuf, loaded_address: u64) -> Result<GsymResolver, Error> {
+        let mut fo = File::open(&file_name)?;
```

Comment:
I will open an issue to handle it.  I think it would be better to handle the error in `ResolverMap::new()`, which is easier to provide an error message in the terminologies used by APIs.